packages/babel-helpers/src/helpers.js

```diff
@@ -4,9 +4,9 @@ let helpers = {};
 export default helpers;
 
 helpers.typeof = template(`
-  (function (obj) {
-    return obj && typeof Symbol !== "undefined" && obj.constructor === Symbol ? "symbol" : typeof obj;
-  });
+  (typeof Symbol === "function" && typeof Symbol() === "symbol")
```

Comment:
@gnarf 
I'm not a pro at Babel's helpers. I based this PR off of the [extends helper](https://github.com/jdalton/babel/blob/76b049f910a91d009adff650c7a5e461cd9f66b4/packages/babel-helpers/src/helpers.js#L165-L177).
The idea is that this is an expression that will be assigned to an identifier. So the end result would look something like:
```js
var _typeof = (typeof Symbol === "function" && typeof Symbol() === "symbol")
  ? function (obj) { return typeof obj; }
  : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol ? "symbol" : typeof obj; };
```
So for modern engines the `_typeof` helper would be just `function (obj) { return typeof obj; }`.